packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -396,10 +403,13 @@ This component is dependent on the [`tiptap`](https://tiptap.dev/) third party l
 library. For the currently supported features, we will include the following libraries that will be added to the package.json
 
 -   [@tiptap/core](https://www.npmjs.com/package/@tiptap/core)
--   [@tiptap/pm](https://www.npmjs.com/package/@tiptap/pm)
 -   [@tiptap/starter-kit](https://www.npmjs.com/package/@tiptap/starter-kit)
+-   [@tiptap/extension-placeholder](https://www.npmjs.com/package/@tiptap/extension-placeholder)
+-   [@tiptap/extension-link](https://www.npmjs.com/package/@tiptap/extension-link)
```

Comment:
Is the link package required as part of placeholder footer? Why is this being included here?

Reply:
If this PR is adding links support then I have several questions:

- SystemLink has some requirements around external linking behaviors used for custom properties in the table. Being able to restrict what URLs can do, etc. Does that apply here? What configuration is needed? Do we have the right APIs to configure that
- Are there visuals needed on URLs that link externally?
- Are the urls configured securely? (link rel noopener, no referrer etc)
- Do links open in a new window or the current window? Do links need Angular Router integration?
- Does this include the visuals of the link editor? If so can you include a screenshot of what that dialog looks like? Also I'm assuming that dialog will be defined in nimble itself as part of the rich text editor template, is that the implementation plan?
- If the link dialog is defined in nimble, what is the localization impact for the strings in the dialog? Do we need to add additional label provider? https://nimble.ni.dev/storybook/?path=/docs/tokens-label-providers--docs

I don't think the addition of link support as part of the spec was clear to other reviewers. The spec itself still says in the `_Additional features out of scope of this spec_` that links are not supported: "Support for adding hyperlinks to the existing words in the editor.", what exactly is being covered here?

Reply:
I have captured a few pieces of information related to these questions in the spec. I have noted some of them down here, but for a few, we don't have immediate answers. We still need to align with the necessary people to come to a decision.
1. As mentioned in the spec, we will currently support only the absolute URL format, which is one of the checks in the hyperlink support for the custom property table. The other validation involves checking if it is an external link, but we do not have an answer for that yet, and we are unsure about the configuration and API requirements to support it. Presently, the server-side does not perform any validation for links that are inserted into comments.
2. For the initial pass, we do not plan to display any visuals for external links.
3. Yes, Tiptap by default adds the rel attribute with noopener, noreferrer values also provide ways to add/override them.
4. Yes, links will open in a new window by default as they have the attribute `target="_blank"`.
5. Currently, we are not planning to support links that are hidden behind text. We will only support absolute URL links, and in markdown, the `autolink` format will be the corresponding representation of it. Therefore, there will be no dialogue to add links to the selected texts.
6. Since the dialog for the link is not currently supported, we will not have the localization.

> "Support for adding hyperlinks to the existing words in the editor.", what exactly is being covered here?

We will indeed support only absolute URLs where the href and the content are the same, and they will be considered as a link in the editor, as shown in the image below.
![image](https://github.com/ni/nimble/assets/123377523/5ec6e8a0-356d-4bd9-8b9d-e38eb0b0906a)

Please note that the information provided regarding the link support represents our current plans and is not a final decision. However, we would like to hear your thoughts on the same.

Reply:
I think the link stuff should be split into a separate spec update. I don't think the reviewers who have currently approved are clearly aware of the approval of link support.

My concerns / thoughts are:
- Any restrictions / limitations placed on the links that can be inserted beyond what CommonMark defines seem application specific. I am not convinced application specific restrictions should be enabled by default on the rich text control.
  - "Presently, the server-side does not perform any validation for links that are inserted into comments" if it will only be enforced by the client then it does not exist. I'd want to hear more justification about enforcing it client side only.
- I disagree with opening links in a new window by default. If that is a SystemLink comments requirement (which I would also push back on) then it should be a configurable option.
- The point still stands that the Globalization section of the spec needs to be updated. The linked PRs have merged so we need to localize the buttons in the toolbar.

Again to avoid blocking the current changes, I'd suggest a separate spec update for the link support.

Reply:
Thanks for sharing your thoughts! As you said, I have moved all the link related changes to a separate PR and it is still in progress as we need to align the link specific requirements with the necessary people. I will also be covering the Globalization part in the draft PR I have created for link support.